### ISSUE TITLE
redis.client.PubSub().parse_response() calling 

### ISSUE THREAD

**Issue by stacywsmith** [redis/redis-py#2754]
**Version**: 4.5.5
**Platform**: CPython 3.8.16 on Ubuntu 

**Description**: 
[`redis.client.PubSub().parse_response()`](https://github.com/redis/redis-py/blob/v4.5.5/redis/client.py#L1540) is calling [`redis.connection.PythonParser().read_response()`](https://github.com/redis/redis-py/blob/v4.5.5/redis/connection.py#L344) with a non-existent `disconnect_on_error` keyword argument.

```
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/client.py:1690: in get_message
08:24:16     response = self.parse_response(block=(timeout is None), timeout=timeout)
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/client.py:1542: in parse_response
08:24:16     response = self._execute(conn, try_read)
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/client.py:1518: in _execute
08:24:16     return conn.retry.call_with_retry(
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/retry.py:46: in call_with_retry
08:24:16     return do()
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/client.py:1519: in <lambda>
08:24:16     lambda: command(*args, **kwargs),
08:24:16 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
08:24:16 
08:24:16     def try_read():
08:24:16         if not block:
08:24:16             if not conn.can_read(timeout=timeout):
08:24:16                 return None
08:24:16         else:
08:24:16             conn.connect()
08:24:16 >       return conn.read_response(disconnect_on_error=False)
08:24:16 E       TypeError: read_response() got an unexpected keyword argument 'disconnect_on_error'
08:24:16 
08:24:16 .tox/_docker/py/lib/python3.8/site-packages/redis/client.py:1540: TypeError
```

**Comment by dvora-h** [comment]
@kristjanvalur Can you take a look here?

**Comment by kristjanvalur** [comment]
self.connection is supposed to be an redis.Connection object.

**Comment by kristjanvalur** [comment]
Ok, the unit tests don't test PubSub with Sentinel connections.  I am not familiar with Sentinel at all.  I presume then, that this connection comes from a SentinelConnectionPool.

I guess it means we need to add this functionality to Sentinel connections.

It is unfortunate that we have all these different combinations of different redis clients and the test suite does not, or isn't able to, test them all.

**Comment by kristjanvalur** [comment]
If you are able to, can you see if pr https://github.com/redis/redis-py/pull/2756 fixes your problem?

**Comment by kristjanvalur** [comment]
sorry, synchronous version now amended too, forgot about that.

**Comment by kristjanvalur** [comment]
Are you able to patch your code with this PR and test?  I'm afraid that the standard redis-py test suite does not excercise all the possible cluster combinations.

**Comment by kristjanvalur** [comment]
don't worry if you can't, this looks like a straigth-forward enough fix and we ought to get it out pretty fast.

**Comment by msingermann-domino** [comment]
Hi! We hit the same issue.
Is there any ETA on when will this fix be released to 4.5?

**Comment by ns-khp** [comment]
Hello! We too hit the same issue.
Is there any ETA on when will this fix be available?

**Comment by c-dev-acc1** [comment]
I got same problem.

```
[2023-05-28 18:16:27,293: INFO/MainProcess] Connected to sentinel://:**@sentinel.redis:5000//
[2023-05-28 18:16:27,302: INFO/MainProcess] mingle: searching for neighbors
[2023-05-28 18:16:28,319: INFO/MainProcess] mingle: all alone
[2023-05-28 18:16:28,352: WARNING/MainProcess] /usr/local/lib/python3.10/site-packages/celery/fixups/django.py:203: UserWarning: Using settings.DEBUG leads to a memory
            leak, never use this setting in production environments!
  warnings.warn('''Using settings.DEBUG leads to a memory

[2023-05-28 18:16:28,352: INFO/MainProcess] celery@django-bi-7959c5555c-scxh4 ready.
[2023-05-28 18:16:28,355: CRITICAL/MainProcess] Unrecoverable error: TypeError("SentinelManagedConnection.read_response() got an unexpected keyword argument 'disconnect_on_error'")
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/celery/worker/worker.py", line 203, in start
    self.blueprint.start(self)
  File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 365, in start
    return self.obj.start()
  File "/usr/local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 332, in start
    blueprint.start(self)
  File "/usr/local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
    step.start(parent)
  File "/usr/local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 628, in start
    c.loop(*c.loop_args())
  File "/usr/local/lib/python3.10/site-packages/celery/worker/loops.py", line 97, in asynloop
    next(loop)
  File "/usr/local/lib/python3.10/site-packages/kombu/asynchronous/hub.py", line 362, in create_loop
    cb(*cbargs)
  File "/usr/local/lib/python3.10/site-packages/kombu/transport/redis.py", line 1326, in on_readable
    self.cycle.on_readable(fileno)
  File "/usr/local/lib/python3.10/site-packages/kombu/transport/redis.py", line 562, in on_readable
    chan.handlers[type]()
  File "/usr/local/lib/python3.10/site-packages/kombu/transport/redis.py", line 906, in _receive
    ret.append(self._receive_one(c))
  File "/usr/local/lib/python3.10/site-packages/kombu/transport/redis.py", line 916, in _receive_one
    response = c.parse_response()
  File "/usr/local/lib/python3.10/site-packages/redis/client.py", line 1542, in parse_response
    response = self._execute(conn, try_read)
  File "/usr/local/lib/python3.10/site-packages/redis/client.py", line 1518, in _execute
    return conn.retry.call_with_retry(
  File "/usr/local/lib/python3.10/site-packages/redis/retry.py", line 46, in call_with_retry
    return do()
  File "/usr/local/lib/python3.10/site-packages/redis/client.py", line 1519, in <lambda>
    lambda: command(*args, **kwargs),
  File "/usr/local/lib/python3.10/site-packages/redis/client.py", line 1540, in try_read
    return conn.read_response(disconnect_on_error=False)
TypeError: SentinelManagedConnection.read_response() got an unexpected keyword argument 'disconnect_on_error'
```

**Comment by dvora-h** [comment]
Fixed in #2756
